File: solution/0001-0100/0003-Longest-Substring-Without-Repeating-Characters/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution{

    public int lengthOfLongestSubstring(String s) {
        if(s == null || s.length() == 0){
            return 0;
        }
        int ans = 0;
        Map<Character, Integer> map = new HashMap<>();
        int left = 0;
        int right = 0;
        while(right < s.length()){
            char c = s.charAt(right);
            if(map.containsKey(c)){
                left = Math.max(map.get(c), left);
            }
            map.put(c, ++right);
            ans = Math.max(ans, right - left);  
        }
        return ans;    
    }
}

File: solution/0001-0100/0003-Longest-Substring-Without-Repeating-Characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = right = result = 0
        map = {}
        
        while right < len(s):
            c = s[right]
            if c in map:
                left = max(map[c], left)
            result = max(result, right - left + 1)
            right = right + 1
            map[c] = right
        return result